src/ci_workflow/ci_check_manifest_component.py

```diff
@@ -0,0 +1,26 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+
+from build_workflow.build_args import BuildArgs
+from ci_workflow.ci_check import CiCheckDist
+from manifests.build_manifest import BuildManifest
+
+
+class CiCheckManifestComponent(CiCheckDist):
+    class MissingComponentError(Exception):
+        def __init__(self, component, url):
+            super().__init__(f"Missing {component} in {url}.")
+
+    def check(self):
+        for architecture in BuildArgs.SUPPORTED_ARCHITECTURES:
+            url = "/".join([self.component.dist, architecture, "manifest.yml"])
```

Comment:
Can we use `urllib` here to construct the url?

Reply:
I had tried, but that gets confused with slashes (does relative path with two slashes), doesn't allow to join more than 2 arguments, etc. So left it alone.